File: sportsbook-app/betting-service/src/main/java/com/sportsbook/bettingservice/kafka/BetPlacedEvent.java
package com.sportsbook.bettingservice.kafka;

import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class BetPlacedEvent {
    private String id;
    private String playerId;
    private String matchId;
    private String userId;
    private String selection;
    private double stake;
    private double odds;

}

File: sportsbook-app/betting-service/src/main/kotlin/com/sportsbook/bettingservice/kafka/BetPlacedEvent.kt
package com.sportsbook.bettingservice.kafka

data class BetPlacedEvent(
    var id: String? = null,
    var playerId: String? = null,
    var matchId: String? = null,
    var userId: String? = null,
    var selection: String? = null,
    var stake: Double = 0.0,
    var odds: Double = 0.0,
)
